src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/context/IActiveContext.kt

```diff
@@ -23,30 +24,29 @@ import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition
 import io.fabric8.kubernetes.client.KubernetesClient
 import io.fabric8.kubernetes.client.Watch
 import io.fabric8.kubernetes.client.Watcher
+import io.fabric8.openshift.api.model.Project
+import io.fabric8.openshift.client.OpenShiftClient
 import java.net.URL
 
 interface IActiveContext<N: HasMetadata, C: KubernetesClient>: IContext {
 
     companion object Factory {
-        fun create(
+        fun createLazyOpenShift(
             client: ClientAdapter<out KubernetesClient>,
-            observable: IResourceModelObservable
+            modelChange: IResourceModelObservable
         ): IActiveContext<out HasMetadata, out KubernetesClient>? {
             val currentContext = client.config.currentContext ?: return null
-            return if (client.isOpenShift()) {
-                OpenShiftContext(
-                    currentContext,
-                    observable,
-                    client as OSClientAdapter
-                )
-            } else {
-                KubernetesContext(
-                    currentContext,
-                    observable,
-                    client as KubeClientAdapter
-                )
-            }
+            return LazyOpenShiftContext(currentContext, modelChange, client as KubeClientAdapter)
```

Comment:
this is the core of the fix: dont check if the cluster is OpenShift and create a Kubernetes- or OpenShift context. This check is costly and may timeout if the cluster is not reachable.
Create a Kubernetes context that async tries to upgrade to an OpenShift context instead.